packages/engine-core/src/data-proxy/utils/getClientVersion.ts

```diff
@@ -25,21 +24,16 @@ async function _getClientVersion(config: EngineConfig) {
   }
 
   // if it's an integration version, we resolve its data proxy
-  if (suffix === 'integration' || suffix?.startsWith('dev') || clientVersion === '0.0.0') {
+  if (suffix !== undefined || clientVersion === '0.0.0') {
     // we infer the data proxy version from the engine version
     const [version] = engineVersion.split('-') ?? []
     const [major, minor, patch] = version.split('.')
 
-    // if a patch has happened, then we return that version
-    if (patch !== '0') return `${major}.${minor}.${patch}`
+    // we will get the latest existing version for this engine
+    const pkgURL = prismaPkgURL(`<=${major}.${minor}.${patch}`)
```

Comment:
Oh wow, how does this work?
Does is essentially get the closest smaller version?

Reply:
Correct, even if the version is non-existing (eg `unpkg.com/prisma@<=42.42.42/package.json`)